cmd/frontend/db/user_emails.go

```diff
@@ -9,17 +9,25 @@ import (
 	"time"
 
 	"github.com/keegancsmith/sqlf"
+	"github.com/sourcegraph/sourcegraph/internal/conf"
 	"github.com/sourcegraph/sourcegraph/internal/db/dbconn"
 	"github.com/sourcegraph/sourcegraph/internal/db/globalstatedb"
 )
 
 // UserEmail represents a row in the `user_emails` table.
 type UserEmail struct {
-	UserID           int32
-	Email            string
-	CreatedAt        time.Time
-	VerificationCode *string
-	VerifiedAt       *time.Time
+	UserID                 int32
+	Email                  string
+	CreatedAt              time.Time
+	VerificationCode       *string
+	VerifiedAt             *time.Time
+	LastVerificationSentAt *time.Time
+}
+
+// NeedsVerificationCoolDown returns true if the verification cooled down time is behind current time.
+func (email *UserEmail) NeedsVerificationCoolDown() bool {
+	return email.LastVerificationSentAt != nil &&
```

Comment:
This would be easier to read IMO. Also, we need to use UTC. Was that not caught in tests?
```suggestion
	return email.LastVerificationSentAt != nil && time.Now().UTC().Before(email.LastVerficationSentAt.Add(conf.EmailVerifcationCoolDown())
```

Reply:
> Also, we need to use UTC. Was that not caught in tests?

I find other places (e.g. `verified_at`) directly use `now()` function, and our table columns are `timestamp with time zone`, meaning the value scanned (e.g. `LastVerificationSentAt`) from table will have timezone information.

Then, if both `time.Time`s have timezone info, we technically don't need to explicitly call `UTC()` before comparison: https://play.golang.org/p/3I3UO-Cqh4i

Reply:
OK. But in general we should be storing UTC timestamps in the database. If that's not the case, it should definitely be.

Here's some good related reading for you to enjoy another time:
- https://zachholman.com/talk/utc-is-enough-for-everyone-right
- http://www.creativedeletion.com/2015/01/28/falsehoods-programmers-date-time-zones.html
- https://medium.com/@kylekatarnls/always-use-utc-dates-and-times-8a8200ca3164

Reply:
Thanks for the list!

---

Looks like we have many instances of using `now()` function in the SQL: https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+lang:go+%3Dnow%28%29&patternType=literal&case=yes

Do you want me to send a PR to use UTC in above results?

Reply:
Can't we make `now()` return UTC by default? I think the `PGTZ` env var controls this.

Reply:
Oh, good to know, double checked our docs documented this: https://docs.sourcegraph.com/dev/postgresql#initializing-postgresql